crates/client/rpc/src/versions/user/v0_8_0/methods/read/get_storage_proof.rs

```diff
@@ -0,0 +1,202 @@
+            let node = match n {
+                mc_db::ProofNode::Binary { left, right } => MerkleNode::Binary { left, right },
+                mc_db::ProofNode::Edge { child, path } => {
+                    MerkleNode::Edge { child, path: path_to_felt(&path), length: path.len() }
+                }
+            };
+            NodeHashToNodeMappingItem { node_hash, node }
+        })
+        .collect();
+
+    Ok((root_hash, converted_proof))
+}
+
+pub fn get_storage_proof(
+    starknet: &Starknet,
+    block_id: BlockId,
+    class_hashes: Option<Vec<Felt>>,
+    contract_addresses: Option<Vec<Felt>>,
+    contracts_storage_keys: Option<Vec<ContractStorageKeysItem>>,
+) -> RpcResult<GetStorageProofResult> {
+    // Pending block does not have a state root, so always fallbacck to latest.
+    let block_id = match block_id {
+        BlockId::Tag(BlockTag::Pending) => BlockId::Tag(BlockTag::Latest),
+        block_id => block_id,
+    };
+
+    let block_n = starknet
+        .backend
+        .get_block_n(&block_id)
+        .or_internal_server_error("Resolving block number")?
+        .ok_or(StarknetRpcApiError::NoBlocks)?;
+
+    let Some(latest) = starknet.backend.get_latest_block_n().or_internal_server_error("Getting latest block in db")?
+    else {
+        return Err(StarknetRpcApiError::BlockNotFound.into());
+    };
+
+    if latest.saturating_sub(block_n) > starknet.storage_proof_config.max_distance {
+        return Err(StarknetRpcApiError::CannotMakeProofOnOldBlock.into());
+    }
+
+    let block_hash = starknet
+        .backend
+        .get_block_hash(&block_id)
+        .or_internal_server_error("Resolving block hash")?
+        .ok_or(StarknetRpcApiError::NoBlocks)?;
+
+    let class_hashes = class_hashes.unwrap_or_default();
+    let contract_addresses = contract_addresses.unwrap_or_default();
+    let contracts_storage_keys = contracts_storage_keys.unwrap_or_default();
+
+    // Check limits.
+
+    let proof_keys = saturating_sum(
+        iter::once(class_hashes.len())
+            .chain(iter::once(contract_addresses.len()))
+            .chain(contracts_storage_keys.iter().map(|v| v.storage_keys.len())),
+    );
+    if proof_keys > starknet.storage_proof_config.max_keys {
+        return Err(StarknetRpcApiError::ProofLimitExceeded {
+            kind: StorageProofLimit::MaxKeys,
+            limit: starknet.storage_proof_config.max_keys,
+            got: proof_keys,
+        }
+        .into());
+    }
+
+    let n_tries = saturating_sum(
+        iter::once(!class_hashes.is_empty() as usize)
+            .chain(iter::once(!contract_addresses.is_empty() as usize))
+            .chain(contracts_storage_keys.iter().map(|keys| (!keys.storage_keys.is_empty() as usize))),
+    );
+    if n_tries > starknet.storage_proof_config.max_tries {
+        return Err(StarknetRpcApiError::ProofLimitExceeded {
+            kind: StorageProofLimit::MaxUsedTries,
+            limit: starknet.storage_proof_config.max_tries,
```

Comment:
Is there a limit to this? 
Usually several contract storage tries are queried

Reply:
yes, for snos you would just put something very big
these limits are just limits that you put for eg. public rpcs so that they don't get DoSed by having to handle huge requests